include/translated-term.php

```diff
@@ -338,4 +338,120 @@ protected function get_db_infos() {
+			return $args['parent'] ?? 0;
+		};
+
+		// Set term parent and language for suffixed slugs.
+		add_filter( 'pll_inserted_term_language', $set_language_for_term_slug );
+		add_filter( 'pll_inserted_term_parent', $get_inserted_term_parent );
+
+		$term = wp_insert_term( $term, $taxonomy, $args );
+
+		// Clean up!
+		remove_filter( 'pll_inserted_term_parent', $get_inserted_term_parent );
+		remove_filter( 'pll_inserted_term_language', $set_language_for_term_slug );
+
+		if ( is_wp_error( $term ) ) {
+			// Something went wrong!
+			return $term;
+		}
+
+		$this->model->term->set_language( (int) $term['term_id'], $language );
+
+		if ( ! empty( $args['translations'] ) ) {
+			$this->model->term->save_translations(
+				(int) $term['term_id'],
+				array_merge(
+					$args['translations'],
+					array(
+						$language->slug => (int) $term['term_id'],
+					)
+				)
+			);
+		}
+
+		return $term;
+	}
+
+	/**
+	 * Wraps `wp_update_term` with language feature.
+	 *
+	 * @since 3.7
+	 *
+	 * @param int   $term_id The ID of the term.
+	 * @param array $args {
+	 *     Optional. Array of arguments for updating a term.
+	 *
+	 *     @type string       $alias_of    Slug of the term to make this term an alias of.
+	 *                                     Default empty string. Accepts a term slug.
+	 *     @type string       $description The term description. Default empty string.
+	 *     @type int          $parent      The id of the parent term. Default 0.
+	 *     @type string       $slug        The term slug to use. Default empty string.
+	 *     @type PLL_Language $lang        The term language object.
+	 * }
+	 * @return array|WP_Error An array containing the `term_id` and `term_taxonomy_id`,
+	 *                        WP_Error otherwise.                  WP_Error otherwise.
+	 */
+	public function update( int $term_id, array $args = array() ) {
+		$term = get_term( $term_id );
+		if ( ! $term instanceof WP_Term ) {
+			return new WP_Error( 'invalid_term', __( 'Empty Term.', 'polylang' ) );
+		}
+
+		$language = ! empty( $args['lang'] ) ? $this->model->get_language( $args['lang'] ) : null;
+		if ( false === $language ) {
+			return new WP_Error( 'invalid_language', __( 'Please provide a valid language.', 'polylang' ) );
+		}
+
+		$set_language_for_term_slug = function () use ( $language ) {
+			return $language;
+		};
+		$parent                    = $args['parent'] ?? $term->parent;
+		$get_inserted_term_parent  = function () use ( $parent ) {
+			return $parent;
+		};
+
+		// Set term parent and language for suffixed slugs.
+		add_filter( 'pll_inserted_term_language', $set_language_for_term_slug );
+		add_filter( 'pll_inserted_term_parent', $get_inserted_term_parent );
+
+		$term = wp_update_term( $term->term_id, $term->taxonomy, $args );
+
+		// Clean up!
+		remove_filter( 'pll_inserted_term_parent', $get_inserted_term_parent );
+		remove_filter( 'pll_inserted_term_language', $set_language_for_term_slug );
```

Comment:
Most of the code is duplicate from the previous method. I wonder if we can find a way around this.

Reply:
I found one in e5692087.